src/nni_manager/core/nniDataStore.ts

```diff
@@ -160,25 +167,56 @@ class NNIDataStore implements DataStore {
 
     private async getFinalMetricData(trialJobId: string): Promise<any> {
         const metrics: MetricDataRecord[] = await this.getMetricData(trialJobId, 'FINAL');
-        if (metrics.length > 1) {
-            this.log.error(`Found multiple final results for trial job: ${trialJobId}`);
+
+        const multiPhase: boolean | undefined = await this.isMultiPhase();
+
+        if (metrics.length > 1 && !multiPhase) {
+            this.log.error(`Found multiple FINAL results for trial job ${trialJobId}`);
+        }
+
+        return metrics[metrics.length - 1];
```

Comment:
what happen if metrics.length == 0?

Reply:
It returns undefined as expected if metrics.length is 0